api/v2/types_firewalldeployment.go

```diff
@@ -94,3 +104,25 @@ func (f *FirewallDeploymentList) GetItems() []*FirewallDeployment {
 	}
 	return result
 }
+
+// IsUserdataCompatibleWithFirewallController returns false if there is a firewall userdata
+// annotation present on the firewall deployment, which conflicts with the firewall controller
+// version in the firewall spec.
+func (f *FirewallDeployment) IsUserdataCompatibleWithFirewallController() bool {
+	value, ok := f.Annotations[FirewallUserdataCompatibilityAnnotation]
+	if !ok {
+		return true
```

Comment:
This is confusing, from the signature of the func i expected false here

Reply:
Hope I clarified this a bit.